lib/kafka_gen_stage/consumer_logic.ex

```diff
@@ -28,71 +27,76 @@ defmodule KafkaGenStage.ConsumerLogic do
   @spec prepare_dispatch(
           buffered_msgs :: :queue.queue(),
           buffered_demand :: non_neg_integer(),
-          transformer :: (msg_tuple(), term() -> {term(), term()}),
-          transformer_state :: term,
-          bulk_transformer :: ([msg_tuple()] -> [msg_tuple()])
+          bulk_transformer :: ([msg_tuple()] -> [msg_tuple()]),
+          high_wm :: non_neg_integer()
         ) ::
           dispatch()
-  def prepare_dispatch(queue, 0, _transformer, state, _bulk_transformer) do
-    {[], :no_ack, 0, queue, state}
+  def prepare_dispatch(queue, 0, bulk_transformer, high_wm) do
+    {transform_bulk([], bulk_transformer, high_wm), :no_ack, 0, queue}
```

Comment:
Should we send something when the demand is `0`?

Reply:
No. You're right.